### PR TITLE
MINIFICPP-263: Move merge content to an extension. Also remove circul…

### PR DESCRIPTION
**Pull request by phrocker**
…ar dependencies

MINIFICPP-263: Remove boost deps by updating yaml-cpp

MINIFICPP-263: Resolve issues on u14

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: thirdparty/yaml-cpp-yaml-cpp-20171024/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+set(gtest_force_shared_crt ${MSVC_SHARED_RT} CACHE BOOL
```
Comment (apiri) [inline thirdparty/yaml-cpp-yaml-cpp-20171024/test/CMakeLists.txt]: Can we kill the test folder.  We had done this previously and will help keep our LICENSE a little tidier.

**Comment by phrocker** [comment]
@apiri I removed the tests in my last commit.

**Comment by apiri** [comment]
Looks great.  Will get this merged.  Thanks!

**Comment by apiri** [comment]
@phrocker Included the closes phrase in the commit but didn't seem to get picked up here.  If you could please close this when you get the opportunity it would be appreciated.

**Pull request closed without merging**